src/main/ipv-stub/ipv-token.ts

```diff
@@ -20,11 +25,111 @@ export const handler: Handler = async (
   }
 };
 
-function get(event: APIGatewayProxyEvent): APIGatewayProxyResult {
+async function postAsync(
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> {
+  let clientAssertionJwt: string;
+  try {
+    // validateHeadersOrThrow(event.headers)
+    const body = getValidBodyOrThrow(event.body);
+    clientAssertionJwt = body["client_assertion"] as string;
+  } catch (e) {
+    if (e instanceof Error) {
+      return {
+        statusCode: 400,
+        body: JSON.stringify({
+          message: e.message,
+        }),
+      };
+    } else {
+      throw e;
+    }
+  }
+
+  const publicKey = createPublicKey(process.env.IPV_TOKEN_PUBLIC_KEY as string);
+
+  try {
+    await jwtVerify(clientAssertionJwt, publicKey, {
+      audience: process.env.AUTH_FRONTEND_BASE_URL,
+      subject: "authOrchestrator",
+      issuer: "authOrchestrator",
+    });
+  } catch (e) {
+    return {
+      statusCode: 500,
+      body: JSON.stringify({
+        message: "Failed to verify JWT",
+      }),
+    };
+  }
+
+  return {
+    statusCode: 302,
+    body: JSON.stringify(generateToken()),
+  };
+}
+
+// function validateHeadersOrThrow(headers: APIGatewayProxyEventHeaders) : void {
+//   if (headers["Content-type"] != "application/x-www-form-urlencoded; charset=UTF-8") {
+//     throw new Error(`Unexpected content type header ${headers["content-type"]}`);
+//   }
+// }
+
+function getValidBodyOrThrow(body: string | null): querystring.ParsedUrlQuery {
+  if (body == null) {
+    throw new Error("Missing request body");
+  }
+
+  const query = querystring.parse(body);
+
+  const grantType = query["grant_type"];
+  if (grantType != "authorization_code") {
+    throw new Error("Unexpected grant type (" + grantType + ") in query");
+  }
+
+  const clientAssertionType = query["client_assertion_type"];
+  if (
+    clientAssertionType !=
+    "urn:ietf:params:oauth:client-assertion-type:jwt-bearer"
+  ) {
+    throw new Error(
+      "Unexpected client assertion type (" + clientAssertionType + ") in query"
+    );
+  }
+
+  const authCode = query["code"];
+  if (!(typeof authCode == "string") || authCode == "") {
```

Comment:
At the moment this will accept an auth code with any value. I've hard-coded an auth code of "12345" in ipv-authorize which we can use for now

Reply:
Moved dummy values into a single file.